File: backend/core/src/main/java/site/timecapsulearchive/core/domain/capsule/treasure_capsule/data/dto/TreasureCapsuleSummaryDto.java
package site.timecapsulearchive.core.domain.capsule.treasure_capsule.data.dto;

import java.time.ZonedDateTime;
import java.util.function.Function;
import lombok.Builder;
import site.timecapsulearchive.core.domain.capsule.treasure_capsule.data.response.TreasureCapsuleSummaryResponse;

@Builder
public record TreasureCapsuleSummaryDto(
    String skinUrl,
    ZonedDateTime dueDate,
    String address,
    String roadName

) {

    public TreasureCapsuleSummaryResponse toResponse(final Function<String, String> s3PreSignedUrlForGet) {
        final String treasureSkinUrl = s3PreSignedUrlForGet.apply(skinUrl);

        return TreasureCapsuleSummaryResponse.builder()
            .skinUrl(treasureSkinUrl)
            .dueDate(dueDate)
            .address(address)
            .roadName(roadName)
            .build();
    }
}

File: frontend/ARchive/app/src/main/java/com/droidblossom/archive/data/dto/treasure/response/TreasureCapsuleSummaryDto.kt
package com.droidblossom.archive.data.dto.treasure.response

import com.droidblossom.archive.domain.model.treasure.response.TreasureCapsuleSummary

data class TreasureCapsuleSummaryDto(
    val skinUrl: String,
    val dueDate: String,
    val address: String,
    val roadName: String?,
) {
    fun toModel() = TreasureCapsuleSummary(
        skinUrl = this.skinUrl,
        dueDate = this.dueDate,
        address = this.address,
        roadName = if (this.roadName == "null" || this.roadName.isNullOrEmpty()) "" else this.roadName
    )
}